block/validate.go

```diff
@@ -60,3 +68,19 @@ func (m *Manager) SettlementValidateLoop(ctx context.Context) error {
 		}
 	}
 }
+
+// LoadDrsHistoryFromStore loads drs history from store
+func (m *Manager) LoadDrsHistoryFromStore() error {
+	drsHistory, err := m.Store.LoadDRSVersionHistory()
+	if errors.Is(err, gerrc.ErrNotFound) {
+		m.logger.Info("failed to find drs history in the store, creating new")
+		m.DRSVersionHistory = &types.DRSVersionHistory{}
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("get drs version history: %w", err)
+	}
+
+	m.DRSVersionHistory = drsHistory
```

Comment:
as to my other comment on the applyBlock - not sure the optimization of saving it in memory and than only saving the uniqe changes in store (which I'm not sure is actually the case) is necessary for optimization. 
I'd simply just save the DRS per height and also get rid of the new manager memeber (i.e `DRSVersionHistory`)

Reply:
removed